### ISSUE TITLE
More options?

### ISSUE THREAD

**Issue by CCGLLC-knc** [Dimencia/DU-Industry-Tool#16]
Kindly request the following options:

1) Exclude industry cost (so that the full cost of the required industry is not factored into the cost of a single output unit)
2) "bulk price" - Figure cost to make 1000 then divide by 1000 - smooths out cost of industry and schematics
3) Allow for manual ore price input (presuming this isn't magic with dynamic real-time pricing)
4) Allow for skill levels to be input

Goal:  Create a tool that can be used by an industrialist to figure a fair price for their products.

Nice to have:  Time spent making components - helps budding industrialist optimize initial infrastructure investments for greatest time savings.

Inspiration:  https://du-craft.net/  (now an abandoned project)

**Comment by CCGLLC-knc** [comment]
Request 1 may not be a real thing.  Issue #15 may just be making it look so.  I used a Basic Mining Unit L as a basis, which came in at over 1M quanta, so assumed the cost of the L assembly line was included.

**Comment by tobitege** [comment]
> Request 1 may not be a real thing. Issue #15 may just be making it look so. I used a Basic Mining Unit L as a basis, which came in at over 1M quanta, so assumed the cost of the L assembly line was included.

Yes, I need to fix some wrong factoring going on there. Industry cost itself is never part of any calculation.
Btw, ore prices as well as talents can already be configured (click File button to see menu items for those).

**Comment by Dimencia** [comment]
Bugs have been fixed; 1 is not a thing I think, 3 and 4 already exist, 2 is sort of up to you to do if you want; I think it's more useful to have an exact cost that you can multiply by 1000 if you want

**Comment by CCGLLC-knc** [comment]
Re:  #2 - Was concerned that some tools calculate cost for a batch, and assign that as the unit cost, and may also calculate in the cost of a schematic, which often has multiple runs - thus overpricing something.  Those factors disappear (for all intents), if you calculate for, say, 1000 units, then divide.

For instance, looking at 0.8.0, a Basic Mining Unit L requires at Tier 1 L element schematic at a cost of 30,000 (current server.  But the schematic is good for making 2 units.  0.8.0 shows a cost of 19,680 (?)

**Comment by Dimencia** [comment]
Unsure about schematics, but we calculate the cost of things as if batching didn't exist.  Batching is covered in the factory breakdown export, sort of in a vague way, but it's basically assumed that batching problems are negligible if you're producing in any bulk

And if schematic costs are being applied only once, or for every run, either way sort of contaminates the result even if you average it out.  I'll have to check with Tobi to see how those work, hopefully it applies the costs incrementally based on the number of uses and quantity

**Comment by CCGLLC-knc** [comment]
We crossed in the Ether while I was updating my post.  As long as batch counts are right, I'm good.  Schematics are now a much smaller factor then my previous experience with them back at 0.23.

**Comment by tobitege** [comment]
> Re: #2 - Was concerned that some tools calculate cost for a batch, and assign that as the unit cost, and may also calculate in the cost of a schematic, which often has multiple runs - thus overpricing something. Those factors disappear (for all intents), if you calculate for, say, 1000 units, then divide.
> 
> For instance, looking at 0.8.0, a Basic Mining Unit L requires at Tier 1 L element schematic at a cost of 30,000 (current server. But the schematic is good for making 2 units. 0.8.0 shows a cost of 19,680 (?)

In the calculation you can see these 2 lines:
Item schematic(s) T1EL: 15,000.00q
and
T1P         x27            4,860.00q (00.00:30:00)
The 19,680 is the total of all schematics, the end product's and the schematics involved in the production of it.
Here: 27 schematics for T1 pure products are required, resulting in 4,860q.

**Comment by CCGLLC-knc** [comment]
Ah... very cool.  It goes deeper than I expected.  Thanks for the explanation.
